core/templates/pages/release-coordinator-page/features-tab/features-tab.component.ts

```diff
@@ -129,25 +127,26 @@ export class AdminFeaturesTabComponent implements OnInit {
   // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
   featureFlagNameToBackupMap!: Map<string, PlatformParameter>;
   featureFlags: PlatformParameter[] = [];
-
+  loadingScreenIsShown: boolean = false;
   isDummyApiEnabled: boolean = false;
+  loadingMessage: string = '';
+  directiveSubscriptions = new Subscription();
 
   constructor(
     private windowRef: WindowRef,
-    private adminDataService: AdminDataService,
-    private adminTaskManager: AdminTaskManagerService,
     private apiService: PlatformFeatureAdminBackendApiService,
     private featureService: PlatformFeatureService,
     private dummyApiService: PlatformFeatureDummyBackendApiService,
+    private loaderService: LoaderService,
   ) {}
 
   async reloadFeatureFlagsAsync(): Promise<void> {
-    const data = await this.adminDataService.getDataAsync();
-
+    const data = await this.apiService.getFeatureFlags();
+    this.loadingScreenIsShown = false;
```

Comment:
Why do you need both this and hideLoadingScreen() below?

Reply:
But semantically, "this.loadingScreenIsShown = false" and "this.loaderService.hideLoadingScreen()" sound like they are doing similar things. So I'm asking why we need to call both of them. What does each contribute? I think you justified the first but not the second.

If they are doing different things, can you find a way to differentiate the names so that it's clear what role each plays?

Also by "check" in your previous comment, do you mean "assignment"? It doesn't seem like a check to me, just a direct variable assignment.

Reply:
I see. I have renamed the variable to `featureFlagsAreFetched`.
So, `loaderService.showLoadingScreen('Loading')` and `loaderService.hideLoadingScreen()` is used to show and hide the loading screen. The `featureFlagsAreFetched` variable is used to keep track of whether the feature flags are fetched completely from the backend or not, So now the `No platform features are available` part will only appear when the `featureFlagsAreFetched` variable is set to `true` and the `featureFlags.length === 0`.

> Also by "check" in your previous comment, do you mean "assignment"? It doesn't seem like a check to me, just a direct variable assignment.

By check, I was referring to the `*ngIf` we have on the HTML page. I was basically referring to the removal of the variable.